src/components/ApiAction.vue

```diff
@@ -0,0 +1,416 @@
+        }
+        return q.name === name
+      })
+      const nameAlreadyUsed = nameSearch !== undefined
+      return !nameAlreadyUsed
+    },
+    storeNewQuery(name) {
+      this.tabs[this.newSaveTabIdx].name = name
+      this.tabs[this.newSaveTabIdx].saved = true
+      this.tabs[this.newSaveTabIdx].savedIdx = this.savedQueries.length
+      const tab = JSON.parse(JSON.stringify(this.tabs[this.newSaveTabIdx]))
+      this.savedQueries.push(tab)
+      this.newSaveTabIdx = null
+      this.storeQueriesToLocalStorage()
+    },
+    saveQuery(tabIdx) {
+      const tab = this.tabs[tabIdx]
+      const storedQueryIdx = this.savedQueries.findIndex(
+        q => q.name === tab.name
+      )
+      if (storedQueryIdx !== -1) {
+        this.savedQueries[storedQueryIdx].query = JSON.parse(
+          JSON.stringify(tab.query)
+        )
+        this.tabs[tabIdx].saved = true
+        this.storeQueriesToLocalStorage()
+      } else {
+        this.newSaveTabIdx = tabIdx
+        this.$bvModal.show('modal-save-query')
+      }
+    },
+    loadStoredQueriesFromLocalStorage() {
+      const storedQueries = localStorage.getItem('storedQueries')
+      if (!storedQueries) {
+        return
+      }
+      const jsonStoreQueries = JSON.parse(storedQueries)
+      if (!jsonStoreQueries[this.currentEnvironment.name]) {
+        return
+      }
+      this.savedQueries = jsonStoreQueries[this.currentEnvironment.name].map(
+        (q, idx) => {
+          q.response = ''
+          q.saved = true
+          q.savedIdx = idx
+          return q
+        }
+      )
+    },
+    storeQueriesToLocalStorage() {
+      let storedQueries = JSON.parse(localStorage.getItem('storedQueries'))
+      if (!storedQueries) {
+        storedQueries = {}
+      }
+      const queriesToStore = JSON.parse(JSON.stringify(this.savedQueries))
+      storedQueries[this.currentEnvironment.name] = queriesToStore.map(q => {
+        delete q.response
+        delete q.saved
+        delete q.savedIdx
+        return q
+      })
+      localStorage.setItem('storedQueries', JSON.stringify(storedQueries))
+    },
+    async performQuery(tabIdx) {
+      const query = JSON.parse(JSON.stringify(this.tabs[tabIdx].query))
+      let response = {}
+
+      try {
+        response = await this.$kuzzle.query(query)
+      } catch (error) {
+        response = {
+          ...error,
+          message: error.message,
+          stack: error.stack,
+          kuzzleStack: error.kuzzleStack
+        }
+      }
+      this.tabs[tabIdx].response = response
+    },
+    newTab() {
+      let newTab = JSON.parse(JSON.stringify(this.emptyTab))
+      this.tabs.push(newTab)
+      this.currentTabIdx = this.tabs.length - 1
+    },
+    async getKuzzleOpenApi() {
+      try {
+        const openApi = await this.$kuzzle.query({
+          controller: 'server',
+          action: 'openapi'
+        })
+        this.openapi = openApi.paths
+      } catch (error) {
+        this.$log.error(error)
```

Comment:
No toast?

Reply:
Nope, we decided to remove the toasts from this view because the response card already display the error and has the error variant ...

Reply:
But this error is not shown in the response view, is it?